packages/react-app/src/components/giveaways/MainContent.tsx

```diff
@@ -18,6 +22,19 @@ const GiveawayMainContent = () => {
   const nrParticipants = giveaway.nrConfirmedParticipants;
   const nrPending = giveaway.nrPendingParticipants;
 
+  const winningChance = useMemo(() => {
```

Comment:
Let's add this calculation to the stats in the API

Reply:
This property depends on which user is making the request. The probability changes if the user is already registered on the giveaway. This can be implemented after authentication is setup on the backend.

Reply:
In that case I suggest refactoring the function so it is easier to understand the calculation. We should also account for the number of possible winners in the calculation since that might not always be 1. Here is a sugestion:

```javascript
function winningChance() {
  if (nrParticipants === 0) return 100;
  
  const isRegistered = participants.some((p) => p.id === userInfo?.email && p.state === 'confirmed');
  const totalParticipants = isRegistered ? nrParticipants : nrParticipants + 1;
  const winningChance = Math.floor((giveaway.numberOfWinners / totalParticipants) * 100);

  return winningChance;
}
```